### PR TITLE
View users logins

### PR DESCRIPTION
**Pull request by TravisMac**

### CODE REVIEW

File: src/forms.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class submit_access_form(forms.Form):
```
Comment (thurloat) [inline src/forms.py]: Style:

classes should be uppsercased. `class SubmitAccessForm(forms.Form):`

for a great guide of consisten python style, check out PEP8 http://www.python.org/dev/peps/pep-0008/

File: src/forms.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+    selection = forms.ChoiceField(choices=[])
+
+    def __init__(self, list_of_orgs, *args, **kwargs):
+        super(submit_access_form,self).__init__(*args, **kwargs)
+        self.fields['selection'].choices = [(org.pk,org.name) for org in list_of_orgs]
```
Comment (thurloat) [inline src/forms.py]: check the `Whitespace in Expressions and Statements` section in PEP8 for making things a bit more standard here.

```
super(SubmitAccessForm, self).__init__(*args, **kwargs)
self.fields['selection'].choices = [ org.pk, org.name for org in list_of_orgs]
```

especially note the spaces after commas when providing args to a call
as well, not that in the list comprehension, putting a comma between the two variables implies a tuple

File: src/settings.py [hunk 3]
```diff
@@ -1,8 +1,11 @@
 # Django settings for access_manager project.
+import os
 
-DEBUG = True 
+DEBUG = False
```
Comment (thurloat) [inline src/settings.py:4]: During development, `DEBUG` should stay on so you get nice stack traces

File: src/ssheepdog/static/css/base.css [hunk 4]
```diff
@@ -0,0 +1,35 @@
+table.access_view {
+    clear: left;
+    border: 1px solid #ddd;
+    border-style:solid;
+    border-width:1px;
+    margin: 6px 1% 10px 0;
```
Comment (thurloat) [inline src/ssheepdog/static/css/base.css]: it's generally bad style to mix pixels with percentages in your CSS.

generally, stick to `px`, `em`, or `%` for units throughout your margins, paddings and widths

File: src/ssheepdog/views.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
+from django.http import HttpResponse
+from django.contrib.auth.models import User
+from django.template import Context, loader
+from src import ssheepdog
+from ssheepdog.models import UserProfile, Machine, Login, Client
+from django.shortcuts import render_to_response
+from django.template import RequestContext
+
+def view_page(request):
```
Comment (thurloat) [inline src/ssheepdog/views.py:9]: You should have some documentation for this view.

```
def view_page(request):
    """
    This view shows xxx, and yyy. 
    foo and bar are special about it.
    """
    ...
```

so others working on the project know what the view is supposed to do without having to read through each line.

File: src/ssheepdog/views.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    #user_prof = get_object_or_404(UserProfile,pk = user_id)
+    #login = get_object_or_404(UserProfile,pk = user_id)
+    #context_dict = {'user' : user_prof,'login' : login}
+    profile = User.objects.select_related('_profile_cache')
```
Comment (thurloat) [inline src/ssheepdog/views.py]: why select all profiles via the User object?

Best practice for using the `select_related()` is to enhance your recordset.

so you would bolt the profile onto the user object without requiring an additional query.

```
users = User.objects.all().select_related('_profile_cache')
```

will allow to you access the profile right on the user object. 

### Edit

Upon further review, this variable isn't being used.

Didn't read enough of the code first. I noticed that down on line ~24 you're making a call to `.get_profile()` which is where the `.select_related()` call can reduce the amount of queries you're doing.

I think you're prematurely optimizing here without clearly understanding the benefits and drawbacks of what's going on.

File: src/ssheepdog/views.py [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+def view_page(request):
+    #user_prof = get_object_or_404(UserProfile,pk = user_id)
+    #login = get_object_or_404(UserProfile,pk = user_id)
+    #context_dict = {'user' : user_prof,'login' : login}
+    profile = User.objects.select_related('_profile_cache')  
+    users = User.objects.all()
+    user_profiles = UserProfile.objects.all()
+    logins = Login.objects.all()
+    for login in logins:
+        login.entries = []
```
Comment (thurloat) [inline src/ssheepdog/views.py:15]: I would bolt this functionality onto the Login model so an instance can have entries that you're just not storing in the db.

This way, you can pass a couple args to a function and you won't have to worry about messing up the list of dicts you're building within the view.

Another good reason to move this functionality out is for testability. You have a very testable piece of code hidden within a view that isn't easily unit testable. With this code refactored into the Login model, you'll be able to write fast tests around it to make sure it never breaks.

File: src/ssheepdog/views.py [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+            all_active_bool = False
+            allowed = False
+            if user in login.users.all():
+                allowed = True
+            if user.get_profile().is_active and login.is_active and login.machine.is_active:
+                all_active_bool = True
+            login.entries.append({'all_active': all_active_bool, 
+                                  'is_allowed': allowed,
+                                  'user': user})
+    context_dict = {'users' : users, 'logins' : logins, 'profiles' :
```
Comment (thurloat) [inline src/ssheepdog/views.py]: Style: if a dict is too long to fit on one line, break each pair onto a newline

```
context_dict = {
    'users': users,
    'logins': logins # ... etc
```

File: src/templates/view_grid.html [hunk 9]
```diff
@@ -0,0 +1,30 @@
+{% extends 'base.html' %}
+{% block content %}
+<link rel="stylesheet" href="{{ STATIC_URL }}css/base.css" type="text/css" />
```
Comment (thurloat) [inline src/templates/view_grid.html]: CSS should go in the base template and within the `<head>` tag inside of that.

```
<html>
    <head>
        <-- CSS Here -->
    </head>
   <body>
  ....
```

File: Vagrantfile [hunk 10]
```diff
@@ -0,0 +1,92 @@
+Vagrant::Config.run do |config|
+  config.vm.forward_port "ssh", 22, 2222
+  config.vm.customize do |vm|
+    vm.memory_size = 128
+    vm.name = "My Project VM"
```
Comment (honza) [inline Vagrantfile:5]: `"SsheepDog VM"`?

**Comment by honza** [comment]
Blast from the past...

**Pull request closed without merging**